universal-login-relayer/lib/services/CounterfactualIdentityService.js

```diff
@@ -0,0 +1,125 @@
+import Identity from 'universal-login-contracts/build/Identity';
+import {addressToBytes32} from '../utils/utils';
+import ethers, {utils, Wallet} from 'ethers';
+import defaultDeployOptions from '../config/defaultDeployOptions';
+
+
+class CounterfactualIdentityService {
+  constructor(wallet, ensService, authorisationService, hooks, provider) {
+    this.wallet = wallet;
+    this.abi = Identity.interface;
+    this.ensService = ensService;
+    this.authorisationService = authorisationService;
+    this.codec = new utils.AbiCoder();
+    this.hooks = hooks;
+    this.provider = provider;
+    this.minimumWeiInContractBeforeDeploy = 1;
+    this.deploymentValue = utils.bigNumberify(utils.parseEther('0.4'));
+  }
+
+  async create(managementKey, ensName, overrideOptions = {}) {
+    let deployTransaction = this._createDeployTransaction(managementKey, ensName);
+
+    deployTransaction = this._setupDeployTransaction(deployTransaction, overrideOptions);
+
+    const signedTransaction = this._signDeployTransaction(deployTransaction);
+
+    const counterfactualTransaction = this._counterfactualizeSignedTransaction(signedTransaction);
+
+    const fullCounterfactualData = await this._extractCounterfactualParams(counterfactualTransaction);
+
+    this.hooks.emit('counterfactuallyCreated', {address: fullCounterfactualData.counterfactualContractAddress});
+
+    return fullCounterfactualData;
+  }
+
+  _createDeployTransaction(managementKey, ensName) {
```

Comment:
To be consistent with the project I would propose to remove _ notation from private functions.